src/model.rs

```diff
@@ -677,6 +677,22 @@ pub trait ProblemOrSolving {
     ///
     /// This method panics if the constraint cannot be created in the current state, or if any of the variables are not binary.
     fn add_cons_set_pack(&mut self, vars: Vec<Rc<Variable>>, name: &str) -> Rc<Constraint>;
+    /// Adds a new cardinality constraint to the model with the given variables, cardinality limit, and name.
+    ///
+    /// # Arguments
+    ///
+    /// * `vars` - The binary variables in the constraint.
+    /// * `rhs` - The maximum number of non-zero variables this constraint allows
```

Comment:
I think `rhs` is not very descriptive here maybe `cardinality`  Also, it should be a `usize` too or a `u32`, to prevent users from using a negative cardinality. 
Those should be changed in the other places too.

Reply:
I added the changes in #108